src/main/java/longah/handler/PINHandler.java

```diff
@@ -0,0 +1,150 @@
+package longah.handler;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.Scanner;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.nio.charset.StandardCharsets;
+import java.math.BigInteger;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+/**
+ * Handles the creation, loading, authentication and resetting of the PIN.
+ */
+public class PINHandler {
+    private static Logger logger = Logger.getLogger("PIN Logger");
+    private static final String PIN_FILE_PATH = "./data/pin.txt";
+    private Scanner scanner;
+
+    /**
+     * Constructs a new PINHandler instance.
+     */
+    public PINHandler() {
+        this.scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Returns the file path of the PIN file where the PIN is encrypted and saved.
+     *
+     * @return The file path of the PIN file where the PIN is encrypted and saved.
+     */
+    public static java.lang.String getPinFilePath() {
+        return PIN_FILE_PATH;
+    }
+
+    /**
+     * Creates a new PIN for the user.
+     */
+    public void createPin() {
+        System.out.println("Thanks for choosing LongAh! Never worry about owing money during the Year of the Dragon!\n"
+                + "Create your 6-digit PIN:\n");
+        String pin = scanner.nextLine();
+
+        while (pin.length() != 6 || !pin.matches("\\d{6}")) {
+            if (Objects.equals(pin, "quit")) {
```

Comment:
believe our stop command is `exit` instead

Reply:
resolved